.github/workflows/build-and-publish.yml

```diff
@@ -2,7 +2,7 @@ name: Build and Publish
 
 on:
   push:
-    branches: [ dev , master, v3/longTermBranch ]
```

Comment:
also the create tag step of releases is failing, I believe this is because the workflow is missing permissions.
We might want to add a permission block to the create tag job with content write.
https://github.com/microsoftgraph/msgraph-sdk-java-core/actions/runs/7885345014

Reply:
it's a recent change from our compliance friends that's enforced on both the Microsoft Graph and Microsoft GitHub Organizations https://docs.opensource.microsoft.com/github/apps/permission-changes/

Reply:
Ah okay, thanks for the ref, I'll add the permissions block to the build and publish yml